### PR TITLE
Bootstrap crc in github action

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

``` release-note
NONE
```

### CODE REVIEW

File: .github/workflows/test-on-crc.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      run: |
+        crc config set preset microshift
+        echo "$PULL_SECRET_CONTENT" > pull-secret
+        crc config set pull-secret-file pull-secret
+        crc config set network-mode user
+        crc config set memory 14000
+    - name: Setup the crc
+      run: sudo -su $USER crc setup
+    - name: Start the crc
+      run: sudo -su $USER crc start
```
Comment (vdemeester) [inline .github/workflows/test-on-crc.yaml:64]: Could use `crc start --pull-secret-file pull-secret` I think to make it work

File: .github/workflows/test-on-crc.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
+name: Test Chains on CRC
+on:
+  # only check PRs for new issues until preexisting issues are cleaned
+  pull_request:
+   branches:
+      - main
+  push:
+    branches:
+      - run-e2e-test-using-crc-in-github-actions
```
Comment (vdemeester) [inline .github/workflows/test-on-crc.yaml:9]: @PuneetPunamiya can you change this to `on-push` on `main` branch instead ?

Comment (PuneetPunamiya) [reply .github/workflows/test-on-crc.yaml:9]: Yeah, I'll update the pr with `on-push`

Comment (PuneetPunamiya) [reply .github/workflows/test-on-crc.yaml]: Updated!

**Review by vdemeester — APPROVED** [review]
/lgtm

**Review by chitrangpatel — APPROVED** [review]

**Pull request merged**